Sources/NIOWebSocket/WebSocketProtocolErrorHandler.swift

```diff
@@ -35,10 +35,10 @@ public final class WebSocketProtocolErrorHandler: ChannelInboundHandler {
             context.writeAndFlush(self.wrapOutboundOut(frame)).whenComplete { (_: Result<Void, Error>) in
                 context.close(promise: nil)
             }
-        }
 
-        // Regardless of whether this is an error we want to handle or not, we always
-        // forward the error on to let others see it.
-        context.fireErrorCaught(error)
+            // Regardless of whether this is an error we want to handle or not, we always
+            // forward the error on to let others see it.
+            context.fireErrorCaught(error)
```

Comment:
This change does not look correct. We only forward now when its a `NIOWebSocketError`. I think we should always forward (as also stated in the comment) ;)

Reply:
Please also add a test for this

Reply:
@normanmaurer good catch!

Reply:
@normanmaurer done